directory-structure.md

```diff
@@ -0,0 +1,62 @@
+# React Component File Naming & Directory Structure
+
+The naming of files and the moving of components into separate files has the intent of removing the noise commonly found in React component files.
+
+### React components
+
+Most components should have their own folder titled using Pascal Casing and have the component defined in the `index.js` file for that folder.
+
+* The final component wrapped with any higher order components should be the default export
+* The component without any higher order wrappings should be a named export for testing purposes
+
+```
+`HelloWorld/index.js` // Good
+
+`HelloWorld.js`       // Bad for most components, ok for internal components... see below
+`helloWorld.js`       // Bad for a React component, but good for a non React component JavaScript module
+`hello-world.js`      // Bad
+```
+
+### Internal components
+
+Private internal components used only in their parent component:
+
+* Can be defined inline with their parent component's `index.js` file
+* Can be moved to a separate file under their parent's directory
+
+This is up to the discresion of the developers. If there is any code branching or significant lines of code (20+), then it probably is a good candidate to move to a separate file.
+
+The goal is to eliminate noise in the parent component's file.
+
+### Stylized Components using stranger
```

Comment:
Long term, I'd like to avoid splitting out CSS into a separate file as I'd like to avoid *writing* lots of CSS. By keeping it in the file you feel the pain of too much CSS.

I'd need to refactor something bigger in Abby to get that across, but this is the type of refactoring I'd like to do in the future to avoid too much custom CSS: https://github.com/everydayhero/donor-ui/pull/43/files